workitems/tokenlist/l2-token-list-v1.0-psd01.md

```diff
@@ -621,12 +621,12 @@ This document defines the conformance levels of a canonical token list as follow
 #### **[D3]** 
 A claim that a canonical token list implementation conforms to this specification SHOULD describe the testing procedure used to justify the claim.
 
-[D3]](#d3) testability: Since each of the non-conformance-target requirements in this documents is testable, so must be the totality of the requirements in this document.
+[D3]](#d3) testability: Since each of the non-conformance-target requirements in this documents is testable, so must be the totality of the requirements in this document. Therefore, conformance tests can exist, and can be described as required in [D3]](#d3).
```

Comment:
The requirement isn't that it is possible to write a description - nor does the requirement as stated actually say the procedure described has to cover anything specific.

So the fact that it's possible to describe conformance testing that covers the specification in its entirety isn't relevant to this requirement.

But probably, tweaking the requirement to match what you're explaining is testable would be the right approach here (and _mutatis mutandi_ for R5)...

Reply:
@chaals the requirement asks for a description of the testing procedure that you are using to justify the claim of conformance for each requirement. This means your testing procedure must specify how you go about testing each requirement. Therefore, you have tests since if you did not, you could not describe them. 

Is 
```
Therefore, conformance tests for each requirement can exist, can be described in terms of test inputs, test outputs, and a testing procedure, and subsequently collated into an overall testing procedure for the entire standard as required in [D3]](#d3).
```
better?

Reply:
No, because the problem isn't proving that it is possible to write a set of tests that is relevant, the problem is that the requirement you're describing doesn't say "tests that are relevant" or anything like it. It only says "describe the procedure you used". So 

> I looked at my marketing person who said we should say we conform, so we are

is a terrible procedure, but it satisfies the requirements D3 / R5

Reply:
@chaals It says to describe the procedure to justify the claim of conformance at a given level; 1, 2, or 3. And each conformance level asks for specific test fixtures and tests, and test descriptions in the form of a report. Therefore, a requirement that asks for a description of a test procedure to justify a claim of Level - 1 conformance contains specific test descriptions, etc,. as required by Level-1. Hence, the specific scenario you mention to any similar one is specifically excluded. D2 is merely there to suggest generating a testing procedure that is readily readable rather than just a testing report in JSON with commented code that would in principle be sufficient, at least for a Level-1 claim.

Reply:
@chaals good to close this out?

Reply:
No but I don't think it's hard to fix. The requirement should be that the claim justifies itself, rather than describing a testing process and assuming that justifies the claim.

e.g.
> #### **[D3]** 
> A claim that a canonical token list implementation conforms to this specification SHOULD describe a testing procedure carried out for each requirement to which conformance is claimed, that justifies the claim with respect to that requirement.

(and similar for [R5])

Reply:
Ok. Will do the update. Can live with that @chaals